lib/sensu/client/process.rb

```diff
@@ -213,6 +227,10 @@ def process_check_request(check)
           else
             execute_check_command(check)
           end
+        elsif check.has_key?(:extension)
+          if @extensions.check_exists?(check[:extension])
+            run_check_extension(check)
+          end
         else
           if @extensions.check_exists?(check[:name])
```

Comment:
could even just use `|| @extensions.check_exists?(check[:extension])` here, no need for the other conditional.